client/state/ui/billing-transactions/test/reducer.js

```diff
@@ -0,0 +1,186 @@
+				transactionType: 'past',
+				month: '2018-03',
+				operator: 'before',
+			} );
+			expect( state ).toEqual( { month: '2018-03', operator: 'before' } );
+		} );
+
+		test( 'should update existing filter with new month', () => {
+			const state = date(
+				deepFreeze( {
+					month: '2018-03',
+					operator: 'before',
+				} ),
+				{
+					type: BILLING_TRANSACTIONS_FILTER_SET_MONTH,
+					transactionType: 'past',
+					month: '2018-09',
+					operator: 'equal',
+				}
+			);
+			expect( state ).toEqual( { month: '2018-09', operator: 'equal' } );
+		} );
+
+		test( 'should change the existing filter to newest', () => {
+			const state = date(
+				deepFreeze( {
+					month: '2018-03',
+					operator: 'before',
+				} ),
+				{
+					type: BILLING_TRANSACTIONS_FILTER_SET_NEWEST,
+					transactionType: 'past',
+				}
+			);
+			expect( state ).toBeNull();
+		} );
+	} );
+
+	describe( '#page()', () => {
+		test( 'should return default state', () => {
+			const state = page( undefined, {} );
+			expect( state ).toEqual( 1 );
+		} );
+
+		test( 'should update set the page number', () => {
+			const state = page( 1, {
+				type: BILLING_TRANSACTIONS_FILTER_SET_PAGE,
+				transactionType: 'past',
+				page: 3,
+			} );
+			expect( state ).toEqual( 3 );
+		} );
+
+		test( 'should reset to 1 on app filter change', () => {
+			const state = page( 4, {
+				type: BILLING_TRANSACTIONS_FILTER_SET_APP,
+				transactionType: 'past',
+				app: 'Test app',
+			} );
+			expect( state ).toEqual( 1 );
+		} );
+
+		test( 'should reset to 1 on date filter being set to a month', () => {
+			const state = page( 4, {
+				type: BILLING_TRANSACTIONS_FILTER_SET_MONTH,
+				transactionType: 'past',
+				month: '2018-03',
+				operator: 'equal',
+			} );
+			expect( state ).toEqual( 1 );
+		} );
+
+		test( 'should reset to 1 on date filter being set to newest', () => {
+			const state = page( 4, {
+				type: BILLING_TRANSACTIONS_FILTER_SET_NEWEST,
+				transactionType: 'past',
+			} );
+			expect( state ).toEqual( 1 );
+		} );
+
+		test( 'should reset to 1 on search query change', () => {
+			const state = page( 4, {
+				type: BILLING_TRANSACTIONS_FILTER_SET_QUERY,
+				transactionType: 'past',
+				query: 'search query',
+			} );
+			expect( state ).toEqual( 1 );
+		} );
+	} );
+
+	describe( '#query()', () => {
+		test( 'should return default state', () => {
+			const state = query( undefined, {} );
+			expect( state ).toEqual( '' );
+		} );
+
+		test( 'should update from empty query', () => {
+			const state = query( '', {
+				type: BILLING_TRANSACTIONS_FILTER_SET_QUERY,
+				transactionType: 'past',
+				query: 'Test query',
+			} );
+			expect( state ).toEqual( 'Test query' );
+		} );
+
+		test( 'should update existing query', () => {
+			const state = query( 'Test query', {
+				type: BILLING_TRANSACTIONS_FILTER_SET_QUERY,
+				transactionType: 'past',
+				query: 'Another test query',
+			} );
+			expect( state ).toEqual( 'Another test query' );
+		} );
```

Comment:
If you're adding tests for clearing the other reducers (although IMHO these don't add a lot of value, as they're similar to the previous test above them), would you like to add one here too?